### DISCUSSION TITLE
HyperSPI Setup Question

### DISCUSSION THREAD

**Discussion by crmmailman** [awawa-dev/HyperSPI#36] [General]
Good Evening and Merry Christmas!

First off what a great project HyperHDR and HyperSPI are, I was thrilled at the idea that I could work on a cool project that will save me $100s of dollars by not having to buy that extremely expensive phillips hue setup. 

I have the following currently:

- Raspberry Pi 3A+
- WS2812B LED Light Strip
- ESP32

I have my Raspberry Pi (running V20beta) and leds all set up. Now I want to get HyperSPI set up, I'm trying to follow the guide but admittedly I'm a noob at all of this and could use some help. Looking at it I know I need to purchase a 3v to 5v shifter but I'm wondering if I need to do any soldering? I've never soldered before and I don't want to if I don't have to. Is there a solution/way to set up the ESP32 with the volt shifter without soldering?

Thank you so much for your help and please let me know if there is any other information I need to provide.

**Comment by awawa-dev** [comment]
Hi
One of the golden rules is: if it ain't broke, don't fix it ;) Are you having problems with the current system or are there other reasons you want to try a different approach? As for boards that have an integrated level shifter, these are from Adafruit but based on Pico/rp2040 not ESP: you can read about them on the HyperSerialPico website. Support for them in HyperSPI is at the testing stage, the PR has already been submitted here, but it still needs to be checked on these specific boards as soon as I find more time.

**Comment by crmmailman** [comment]
Oh sorry there must be some misunderstanding. This is my first setup I tried the setup with the ws281x controller but the controller does not recognize the hardware and won’t turn on the leds. So now I am doing what I should’ve done in the first place and go the HyperSPI route.

I bought a ESP32 I’m just not all the sure on how I connect it to a level shifter. Is soldering required or is there a way to only do it with jumper wire? Like I said I’m a little scared to attempt soldering for the first time.

On that note, what level shifter is recommended for WS2812Bs?

Thank you for your assistance.


On Dec 24, 2023, at 6:19 AM, Awawa ***@***.***> wrote:

﻿

Hi
One of the golden rules is: if it ain't broke, don't fix it ;) Are you having problems with the current system or are there other reasons you want to try a different approach? As for boards that have an integrated level shifter, these are from Adafruit but based on Pico/rp2040 not ESP: you can read about them on the HyperSerialPico website. Support for them in HyperSPI is at the testing stage, the PR has already been reported here, but it still needs to be checked on these specific boards as soon as I find more time.

**Comment by awawa-dev** [comment]
An example level shifter connection is in the wiki: https://github.com/awawa-dev/HyperHDR/wiki/Level-Shifter Connection diagrams for HyperSPI can also be found in the last blog post or even here on the main page. If you don't want to solder anything (besides the LED strip), you have to wait a little longer for the Adafruit Feather RP2040 Scorpio and ItsyBitsy tests to be completed.

**Comment by crmmailman** [comment]
Thanks! I’ll start learning how to solder.

On Dec 24, 2023, at 2:10 PM, Awawa ***@***.***> wrote:

﻿

An example level shifter connection is in the wiki: https://github.com/awawa-dev/HyperHDR/wiki/Level-Shifter Connection diagrams for HyperSPI can also be found in the last blog post or even here on the main page. If you don't want to solder anything, you have to wait a little longer for the Adafruit Feather RP2040 Scorpio and ItsyBitsy tests to be completed.

**Comment by awawa-dev** [comment]
Without soldering, you can still use an Arduino proto-board something like this to construct the level shifter:
![Bez nazwy](https://github.com/awawa-dev/HyperSPI/assets/69086569/94fe54b7-38fb-4123-8647-2832c4ac1b92)
Just don't run the main power supply for the LEDs through this board because it won't hold it, provide only the data signal and common ground.

Soldering shouldn't be difficult, especially if you use tin lead and a good flux. Just avoid two main mistakes that beginners often make, which makes the task very difficult: you apply the tin directly to the element heated by the soldering iron, not tin to the soldering iron and then to the element, and always pre-tin the wires you want to solder.
